### DISCUSSION TITLE
unable to deploy on vercel although it's working on lacalhost3000

### DISCUSSION THREAD

**Discussion by DarkoKrezic** [neuefische/questions#604] [Web]
Gerade arbeite ich daran meine Einträge zu editieren. 
[Hier geht es zu dem Branch ](https://github.com/DarkoKrezic/capstone-projekt/tree/cloudinary) . Auf dem localen Server läuft es , allerdings deployt Vercel es nicht .
Beschreibung des Fehlers:
Wie kann man den Fehler reproduzieren?
Direkt beim Starten der App in Vercel, oder auch beim aufrufen von npm run build im Terminal wird
Wo und wann tritt er auf?
Direkt beim Starten des Development Servers.
Was wird erwartet, was passiert?
Ich erwarte, dass meine App ohne Fehlermeldung deployed wird. Es wird aber nicht deployed.
Was sind die Fehlermeldungen?
`Export encountered errors on following paths:
        /stories/[id]/EditStory`
`TypeError: Cannot read properties of undefined (reading 'find')`
`Error occurred prerendering page "/stories/[id]/EditStory". Read more: https://nextjs.org/docs/messages/prerender-error` sindauszüge aus dem dem Block.

**Comment by moonwave99** [comment]
The problem happens even on localhost if you try to directly visit a story edit page, e.g. http://localhost:3000/stories/1/EditStory

Solution: shield your check with a `?.` and an if statement like:

```js
const storyToEdit = stories?.find((story) => story.id === parseInt(id));
if (!storyToEdit) {
  return null;
}
```

**Comment by DarkoKrezic** [comment]
well actually its runnin in localhost, but even with shielding i still get the same error
